050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,717 @@
+
+To avoid this it is **recommended** that the Topic Operator not be used with cluster where `auto.create.topics.enable` is `true`. 
+The operator will emit a `WARN` level log on startup if autocreation is enabled. 
+
+Alternatively: App could be written to wait for topic existency, or use an init container to do the same.
+
+## Affected/not affected projects
+
+* The existing bidirectional Topic Operator would be deprecated and removed once ZooKeeper-based clusters are no longer supported.
+* A new unidirectional Topic Operator would be provided as a replacement, which users could migrate to before ZooKeeper-based clusters are no longer supported.
+* The schema of `KafkaTopic` would change
+* The schema of `Kafka` would also change, to support deployment of the new operator, but the details of that are not defined in thisproposal.
+* The CRD changes would be relfected in the `api` module.
+
+## Compatibility
+
+// Call out any future or backwards compatibility considerations this proposal has accounted for.
+
+Let's consider each of the public APIs of the BTO in turn and describe the compatiblity story.
+
+### Operator Configuration
+
+#### Standalone case
+
+<table>
+<tr><th>Env var						<th>Status
+<tr><td>STRIMZI_NAMESPACE				<td>Becomes optional, see note 1
+<tr><td>STRIMZI_POLICY_YAML				<td>Added, optional, see note 1
+<tr><td>STRIMZI_RESOURCE_LABELS				<td>Unchanged
+<tr><td>STRIMZI_KAFKA_BOOTSTRAP_SERVERS			<td>Unchanged
+<tr><td>STRIMZI_CLIENT_ID				<td>Unchanged
+<tr><td>STRIMZI_FULL_RECONCILIATION_INTERVAL_MS		<td>Unchanged
+<tr><td>STRIMZI_TLS_ENABLED				<td>Unchanged
+<tr><td>STRIMZI_TRUSTSTORE_LOCATION			<td>Unchanged
+<tr><td>STRIMZI_TRUSTSTORE_PASSWORD			<td>Unchanged
+<tr><td>STRIMZI_KEYSTORE_LOCATION			<td>Unchanged
+<tr><td>STRIMZI_KEYSTORE_PASSWORD			<td>Unchanged
+<tr><td>STRIMZI_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM	<td>Unchanged
+<tr><td>STRIMZI_SASL_ENABLED				<td>Unchanged
+<tr><td>STRIMZI_SASL_MECHANISM				<td>Unchanged
+<tr><td>STRIMZI_SASL_USERNAME				<td>Unchanged
+<tr><td>STRIMZI_SASL_PASSWORD				<td>Unchanged
+<tr><td>STRIMZI_SECURITY_PROTOCOL			<td>Unchanged
+<tr><td>STRIMZI_ZOOKEEPER_CONNECT			<td>Dropped
+<tr><td>STRIMZI_ZOOKEEPER_SESSION_TIMEOUT_MS		<td>Dropped
+<tr><td>TC_ZK_CONNECTION_TIMEOUT_MS			<td>Dropped
+<tr><td>STRIMZI_REASSIGN_THROTTLE			<td>Dropped, see note 2
+<tr><td>STRIMZI_REASSIGN_VERIFY_INTERVAL_MS		<td>Dropped, see note 2
+<tr><td>STRIMZI_TOPIC_METADATA_MAX_ATTEMPTS		<td>Dropped
+<tr><td>STRIMZI_TOPICS_PATH				<td>Dropped
+<tr><td>STRIMZI_STORE_TOPIC				<td>Dropped
+<tr><td>STRIMZI_STORE_NAME				<td>Dropped
+<tr><td>STRIMZI_APPLICATION_ID				<td>Dropped
+<tr><td>STRIMZI_STALE_RESULT_TIMEOUT_MS			<td>Dropped
+<tr><td>STRIMZI_USE_ZOOKEEPER_TOPIC_STORE		<td>Dropped
+</table>
+
+Notes:
+
+1. `STRIMZI_NAMESPACE` is retained for backwards compatibility and configures the operator with a policy allowing all topics in the given namespace.
+   `STRIMZI_POLICY_JSON` configures a policy as JSON string.
```

Comment:
I don't have a strong preference. I went with JSON because I felt a suspicious of env vars containing newlines  which YAML would require (it seems likely that something, somewhere, eventually is going to do something with the significant whitespace).